apps/browser/qml/pages/components/TabView.qml

```diff
@@ -203,10 +200,89 @@ Item {
 
         ]
     }
+
     TabsToolBar {
         id: tabsToolBar
         anchors.bottom: parent.bottom
         onBack: pageStack.pop()
-        onEnterNewTabUrl: root.enterNewTabUrl()
+        onEnterNewTabUrl: tabView.enterNewTabUrl()
+        onOpenMenu: popupMenu.active = true
+    }
+
+    PopUpMenu {
+        id: popupMenu
+
+        signal closeAllTabs
+
+        anchors {
+            right: parent.right
+            rightMargin: Theme.paddingMedium
+            bottom: parent.bottom
+            bottomMargin: Theme.paddingMedium
+        }
```

Comment:
After rebasing this on next you'll need to change this so the menu item fills the page. The margins will be taken care of internally.

Reply:
done